Include/internal/pycore_stackref.h

```diff
@@ -213,6 +225,27 @@ _PyStackRef_FromPyObjectSteal(PyObject *obj)
 }
 #   define PyStackRef_FromPyObjectSteal(obj) _PyStackRef_FromPyObjectSteal(_PyObject_CAST(obj))
 
+static inline bool
+PyStackRef_IsBorrowed(_PyStackRef stackref)
```

Comment:
I think we should get rid of this function, if possible. It's hard to know what it does without reading the implementation, and the fewer of those types of functions we have the better.

I see only two uses:

* `_PyStackRef_NewIfBorrowedOrSteal`, which can be replaced with the existing `PyStackRef_AsStrongReference`.
* The assertion in `_BINARY_OP_INPLACE_ADD_UNICODE`, which can be weakened or even removed if necessary.